.github/workflows/macos.yml

```diff
@@ -0,0 +1,34 @@
+name: "MacOS"
+
+on:
+  push:
+  pull_request:
+
+jobs:
+  build-macos:
+    name: ${{ matrix.os }} Build FreeOrion SDK
+    runs-on: ${{ matrix.os }}
+    strategy:
+      matrix:
+        os: [macos-10.15]
+    env:
+      CMAKE_BUILD_PARALLEL_LEVEL: 2
+      HOMEBREW_NO_AUTO_UPDATE: 1
+    steps:
+      - name: Checkout sources
+        uses: actions/checkout@v3
+      - name: Configure
+        run: |
+          mkdir build
+          cd build
+          cmake -G "Xcode" ..
```

Comment:
The CI build still fails, so you might try removing the `-G "Xcode` part too. As I said, I only do `cmake ..`without specifying any options, and that works perfectly fine on my system.

Judging by the error message of the failed CI build, it has something to do with a new Xcode build system:
```
CMake Error in CMakeLists.txt:
  The custom command generating

    /Users/runner/work/freeorion-sdk/freeorion-sdk/build/libpng-prefix/src/libpng-build/pnglibconf.c

  is attached to multiple targets:

    genvers
    gensym
    genfiles

  but none of these is a common dependency of the other(s).  This is not
  allowed by the Xcode "new build system".
```
Maybe not specifying Xcode as the generator system to use circumvents that problem?